app/scripts/modules/core/src/overrideRegistry/Overrides.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+
+import { CloudProviderRegistry } from 'core/cloudProvider';
+import { OverrideRegistry } from 'core/overrideRegistry';
+
+/**
+ * Registers a component as a replacement for some other component.
+ *
+ * This is a Class Decorator which should be applied to a React Component Class.
+ * The component class will be used instead of the OverridableComponent with the same key.
+ *
+ * If an (optional) cloudProvider is specified, the component override takes effect only for that cloud provider.
+ * If an (optional) cloudProviderVersion is specified, the component override takes effect only for a specific version of that cloud provider.
+ *
+ * @Overrides('overrideKey', "aws", "v2")
+ * class MyOverridingCmp extends React.Component {
+ *   render() { return <h1>Overridden component</h1> }
+ * }
+ */
+export function Overrides(key: string, cloudProvider?: string, cloudProviderVersion?: string) {
```

Comment:
Currently this decorator is used only for "generic component overrides", not cloud provider overrides.  Although not 100% necessary (you can call `overrideRegistry` manually) I feel that this will decorator be easier when teaching people how to override UI components.